diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0737d7f..a5add16 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,21 +23,6 @@ set(IONC_ENABLE_VERBOSE_DEBUG OFF CACHE BOOL "Enable verbose logging for ion-c")
 if (CMAKE_BUILD_TYPE STREQUAL "Release")
    add_compile_definitions(NDEBUG=1)
 elseif (CMAKE_BUILD_TYPE STREQUAL "Debug")
-   include(CheckCXXCompilerFlag)
-   # check_cxx_compiler_flag doesn't just check compiler flags, it also attempts to link the program, which
-   # in this case, also requires sanitizer flags. So we have to abuse CMAKE_REQUIRED_LINK_OPTIONS since that's
-   # the only lever we're given afaict. This does not work with cmake prior to 3.14.
-   list(INSERT CMAKE_REQUIRED_LINK_OPTIONS 0 "-fsanitize=address,undefined")
-   check_cxx_compiler_flag("-g -fsanitize=address,undefined -fno-omit-frame-pointer" UBISAN_OK)
-   list(REMOVE_AT CMAKE_REQUIRED_LINK_OPTIONS 0)
-   if (UBISAN_OK OR IONC_FORCE_SANITIZERS)
-      add_compile_options(
-         -fsanitize=address,undefined
-         -fsanitize-recover=address
-      )
-      add_link_options(-fsanitize=address,undefined -fsanitize-recover=address)
-   endif()
-   add_compile_options(-g -fno-omit-frame-pointer -fno-optimize-sibling-calls)
    if (IONC_ENABLE_VERBOSE_DEBUG)
        add_compile_definitions(DEBUG=1)
    endif()
